adds a codecov example project to the repo #105

RobStallion opened the pull request:
adds a codecov example project to the repo #105

~Still need to add the codecov badge.~

Cleop commented on codecov_example/README.md:
````diff
@@ -0,0 +1,317 @@
+```
+
+As you can see our tests are passing and all is looking good. However, as you
+have probably seen, we do not have a test for our add three function. Now, in
````
```suggestion
have probably seen, we do not have a test for our `add_three` function. Now, in
```

Cleop commented on codecov_example/README.md:
```diff
@@ -0,0 +1,317 @@
+As you can see our tests are passing and all is looking good. However, as you
+have probably seen, we do not have a test for our add three function. Now, in
+our example it's fairly easy to do an eyeball test and see that we are missing
+a test, but imaging you are in a project with 100 functions all spread out over
```
```suggestion
a test, but imagine you are in a project with 100 functions all spread out over
```

RobStallion replied:
Oh gosh. Thanks cleo 👍

Cleop commented on codecov_example/README.md:
```diff
@@ -0,0 +1,317 @@
+a test, but imaging you are in a project with 100 functions all spread out over
+different modules. The eyeball test isn't so reliable anymore.
+
+That's where code coverage tools can come into handy. So let's add one to our
```
```suggestion
That's where code coverage tools can come in handy. So let's add one to our
```

Cleop commented on codecov_example/README.md:
```diff
@@ -0,0 +1,317 @@
+Next, in the terminal, run the command `mix deps.get`. This will add
+`excoveralls` as a dependency to our project. The `only: :test` part of the
+command is to say that we will only be using this dependency in our test
+environment.
```
```suggestion
environment. You should see this in your terminal:
```

Cleop commented on codecov_example/README.md:
```diff
@@ -0,0 +1,382 @@
+
+We are going to get [travis-ci](https://travis-ci.org/) to check our tests
+online for us when we commit our code to github (this section assumes you are
+going to be pushing your code, so if you are nor planning to then feel free to
```
```suggestion
going to be pushing your code, so if you are not planning to then feel free to
```

Cleop commented on codecov_example/README.md:
```diff
@@ -0,0 +1,382 @@
+We are going to get [travis-ci](https://travis-ci.org/) to check our tests
+online for us when we commit our code to github (this section assumes you are
+going to be pushing your code, so if you are nor planning to then feel free to
+skip this it).
```
```suggestion
skip this bit).
```

Cleop commented on codecov_example/README.md:
```diff
@@ -0,0 +1,382 @@
+skip this it).
+
+In order to get travis to run our tests we need to create a `.travis.yml` file.
+I am not going to go into any detail on the travis-ci steps as there in an
```
```suggestion
I am not going to go into any detail on the travis-ci steps as there is an
```

Cleop commented on codecov_example/README.md:
```diff
@@ -0,0 +1,382 @@
+![Build Status](https://travis-ci.org/dwyl/learn-elixir.svg?branch=master)
+![codecov](https://codecov.io/gh/dwyl/learn-elixir/branch/master/graph/badge.svg)
+
+The format of the links are as links that display the badge images are as
```
```suggestion
The format of the links that display the badge images are as
```

Cleop commented on codecov_example/README.md:
```diff
@@ -0,0 +1,382 @@
+![codecov](https://codecov.io/gh/dwyl/learn-elixir/branch/master/graph/badge.svg)
+
+The format of the links are as links that display the badge images are as
+follows...
```
```suggestion
follows (so customise yours with the appropriate details from your project)...
```

RobStallion commented on codecov_example/README.md:
```diff
@@ -0,0 +1,317 @@
+they can help to highlight parts of code that are not tested and may contain
+errors. They also may highlight the fact that some of the code that was written
+is not even needed, for example a clause in a function that can never be
+reached.
```
Really good point. I'll add that for sure thanks 👍

RobStallion commented on codecov_example/README.md:
```diff
@@ -0,0 +1,317 @@
+Now that that's out of the way let's set up a simple mix project and add code
+coverage to it.
+
+## Setting up our new mix project
```
I figured that people would have read through a decent chunk of the learn elixir readme before getting to this readme so they should have some idea of what this means. (Reminds me that I didn't add the link to the readme though so thanks for that haha)

RobStallion commented on codecov_example/README.md:
````diff
@@ -0,0 +1,317 @@
+  end
+```
+
+As you can see, this has added functions that test our `add_one/1` and
````
This readme will be linked in the main readme after the tests. (Again I forgot to add the link 🤦‍♀️. So whoever is reading it will have just read about tests)

RobStallion commented:
Thanks for your review @Cleop. I have updated the readme with a bunch of your suggestions and added committed some of changes directly. Can you please take another look at this when you have time 👍 🙏
